pkg/signals/signal.go

```diff
@@ -40,6 +44,16 @@ func SetupSignalHandler() (stopCh <-chan struct{}) {
 		<-c
 		os.Exit(1) // second signal. Exit directly.
 	}()
+	go func() {
```

Comment:
Could be extracted to a dedicated function. With that `return stop` would be closer to the corresponding context where `stop` is created/used.

Reply:
As the function now does not _one_ thing anymore but two, I vote for the extraction of _both_ functionalities into separate functions.

Reply:
Perfect.